### PR TITLE
Add 5 new strips, fix one old one

### PR DESCRIPTION
**Pull request by ExTechOp**
Added five new strips (with hopefully accurate information and suitable cover + sample images):
- bizarro (Comicskingdom)
- haraldhirmuinen (Helsingin Sanomat)
- keskenkasvuisia (Helsingin Sanomat)
- lassijaleevi (Helsingin Sanomat)
- ~~thefarside (Larson's website thefarside.com)~~

And corrected retrieval of kamalaluonto (now from Keskisuomalainen newspaper), and updated also its information.

Closes issue #103.

### CODE REVIEW

**Comment by ExTechOp** [comment]
And only after submitting this I realized that thefarside does properly retrieve the images, but they do not include the captions, so most of them are rather incomprehensible 😿

**Comment by Olf0** [comment]
Thank you very much!  Now I feel a bit obliged to create a new release, soon.  I hope I can spare the time the next weekend.

WRT "The Far Side": I am a fan of it since being on school exchange in Washington state in the late 1980ies.  Updating DailyComic's metadata for it a few years ago let me finally understand why my "exchange mother" was such a big fan (an infected me): Gary Larson is from Seattle.  **But** [please see PR \#88](https://github.com/sailfishos-applications/daily-comics/pull/88)!
Hence I will delete it from this PR (before merging).

I will try to quickly review the rest of this PR in the next days.

P.S.: Your contributions mean a big uplift for my motivation to maintain DailyComics.  Because the original creator did stall all PR for years (without deciding to merge them or not), I [closed all my PRs on 2023-09-23](https://github.com/tardypad/sailfishos-daily-comics/pulls?q=is%3Apr+is%3Aclosed) and forked DailyComics (in the original sense of "to fork", i.e. "continue this project separately"), after [proposing to collaborate unsuccessfully a few times](https://github.com/tardypad/sailfishos-daily-comics/issues/137).

**Comment by ExTechOp** [comment]
Yes, I concur that The Far Side like this doesn't really work and including it like this doesn't make sense. One future enhancement might be to be able to display text alongside the image — that would also be the place to include the xkcd rollover texts.

I will admit I still don't quite understand what use the program makes of the specified color, I cannot recall having seen any comics or cover images on my phone on anything but a black background? The colors I've specified with these patches were mostly taken from the cover image, assuming this somehow is representative of color scheme.

**Comment by Olf0** [comment]
> I concur that The Far Side like this doesn't really work and including it like this doesn't make sense.

Deleted by [commit bce1dce](https://github.com/sailfishos-applications/daily-comics/pull/108/commits/bce1dcea5b988a0a2d3a3c44ab4f9863a556d379).

> One future enhancement might be to be able to display text alongside the image — that would also be the place to include the xkcd rollover texts.

If you want to avoid this from becoming forgotten, please file a new issue suggesting exactly this (maybe a little bit more fleshed out).
 
> I will admit I still don't quite understand what use the program makes of the specified color, I cannot recall having seen any comics or cover images on my phone on anything but a black background?

When I reworked the README and all to other guidance (e.g. [`info/comic_addition.md`](https://github.com/sailfishos-applications/daily-comics/blob/devel/info/comic_addition.md)) almost a year ago, I tried to retrace all requirements `@tardypad` originally made: Is it a hard or soft requirement, does it make sense at all etc.  While I do not remember the specifics of the requirement, "background colour must no be all white (`#ffffff`) or all black (`#00000`)", I think (i.e. hope) I did that with my usual diligence: I.e. it once made sense to me to keep it in the documentation, IIRC.

> The colors I've specified with these patches were mostly taken from the cover image, assuming this somehow is representative of color scheme.

Correct, this is another (but rather soft) requirement stated in [`info/comic_addition.md`](https://github.com/sailfishos-applications/daily-comics/blob/devel/info/comic_addition.md) (which absolutely makes sense IMO).

> Edit: for example, what should it be for black-and-white comics?

Pick a different background colour. :wink:  E.g. "almost white" (e.g. `#f0f0f0`) / "almost black" (i.e. "dark gray", e.g. `#0f0f0f`).

**Comment by Olf0** [comment]
My four review comments as screenshots, because something is apparently not working with GitHub's "review"-functionality, lately.

As already stated, only the first two contain real questions, the other two are just comments which you surely can also comment on.
1. screenshot_2024-12-02_020144
2. screenshot_2024-12-02_020435
3. screenshot_2024-12-02_020546
4. screenshot_2024-12-02_020619

**Comment by ExTechOp** [comment]
1. Yep, I can fully understand why you prefer that. Go ahead with the change.
2. Also this Finnish language Wikipedia change is fine.
3. I cannot claim to be an expert of javascript, but the "object-oriented" syntax of calling the String method seems most natural here. [Replace replaces the matched regex](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/String/replace) here in the URL with just the string "FREE" without a trailing number, which seems to provide the highest-resolution image.

**Comment by Olf0** [comment]
> 1.

Done.
 
> 2. 

Done.

> 3. I cannot claim to be an expert of javascript, but the "object-oriented" syntax of calling the String method seems most natural here.

That is the Point: C, Shell-scripts, Fortran, Assembler, Modula, Pascal, Basic are all fine for me, but OO means "Ooooh" to me instead of "Object Oriented", i.e. I have (almost) no clue.

> …, which seems to provide the highest-resolution image.

Perfect, as you checked and implemented that; hence it is fine for me.

**Review by Olf0 — COMMENTED** [review]
Fine now.

File: plugins/keskenkasvuisia/info.json [hunk 1]
```diff
@@ -0,0 +1,11 @@
+{
+  "name": "Keskenkasvuisia",
+  "color": "7eb97f",
+  "language": "fi",
+  "authors": [
+      "Aurora Raiskinen",
+      "Jyrki Vainio"
+  ],
+  "homepage": "https://www.instagram.com/keskenkasvuisia/",
```
[the remark this answers is not in the document]
Comment (Olf0) [reply plugins/keskenkasvuisia/info.json]: **`@ExTechOp`**:
> Yep, I can fully understand why you prefer that. Go ahead with the change.

**`@Olf0`**:
Done.

File: plugins/haraldhirmuinen/info.json [hunk 2]
```diff
@@ -0,0 +1,12 @@
+{
+  "name": "Harald Hirmuinen",
+  "color": "e7c5aa",
+  "language": "fi",
+  "authors": [
+      // These are reruns of strips by either of these, both have passed on
```
[the remark this answers is not in the document]
Comment (Olf0) [reply plugins/haraldhirmuinen/info.json:6]: **`@ExTechOp`**:
> I cannot claim to be an expert of javascript, but the "object-oriented" syntax of calling the String method seems most natural here.

File: plugins/kamalaluonto/extract.js [hunk 3]
```diff
@@ -1,7 +1,7 @@
 function(page) {
-    var regex = /data-image="[^"]*\/([^"\/]*).jpg"/;
+    var regex = /<img\s+class="diks-card__image"\s+src="([^"]+)/;
     var match = regex.exec(page);
-    var hash = match[1]
+    var hash = match[1];
 
-    return "http://is.mediadelivery.fi/img/1920/" + hash + ".jpg";
+    return hash.replace(/FREE_\d+/,"FREE");
```
[the remark this answers is not in the document]
Comment (Olf0) [reply plugins/kamalaluonto/extract.js:6]: **`@ExTechOp`**:
> I cannot claim to be an expert of javascript, but the "object-oriented" syntax of calling the String method seems most natural here. [Replace replaces the matched regex](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/String/replace) here in the URL with just the string "FREE" without a trailing number, which seems to provide the highest-resolution image.

File: plugins/lassijaleevi/info.json [hunk 4]
```diff
@@ -0,0 +1,10 @@
+{
+  "name": "Lassi ja Leevi",
+  "color": "ec1d23",
+  "language": "fi",
+  "authors": [
+    "Bill Watterson"
+  ],
+  "homepage": "http://www.calvinandhobbes.com/",
```
[the remark this answers is not in the document]
Comment (Olf0) [reply plugins/lassijaleevi/info.json]: **`@ExTechOp`**:
> This Finnish language Wikipedia change is fine.

**Comment by ExTechOp** [comment]
(unfortunately it seems Helsingin Sanomat just redid their comics section, and oh boy, each landing page is now put together with a maze of dozens of little javascript routines, all looking alike)

**Comment by Olf0** [comment]
> (unfortunately it seems Helsingin Sanomat just redid their comics section, and oh boy, each landing page is now put together with a maze of dozens of little javascript routines, all looking alike)

Sound as if we have been too slow with this. ;-\

I just took a quick look: As the links to the landing pages are still the same, I assume all corresponding extraction scripts have to adapted (again), right?

**Pull request merged**